lib/ngx/re.lua

```diff
@@ -168,6 +152,24 @@ function _M.split(subj, regex, opts, ctx, max, res)
                 break
             end
 
+            if from == to then
+                local old_pos = ctx.pos
+
+                local from2, _, _, err2 = re_split_helper(subj, compiled,
+                                              compile_once, flags, ctx)
```

Comment:
@thibaultcha It looks strange to me to perform a second match in the same location. Why? You said it is "the only way", but I still don't understand it. Will you elaborate?